### PR TITLE
Generate error messages instead of error codes (close #13)

### LINKED ISSUE #13: Hammerhead should generate error texts instead of codes
\cc @LavrovArtem @miherlosev 

Based on https://github.com/superroma/testcafe-hammerhead/issues/5

When the hammerhead generates an exception it should transmits error text instead of error code.

### PR DESCRIPTION
**Pull request by churkin**
\cc @inikulin @LavrovArtem @kirovboris  

@VasilyStrelyaev check error messages please.

### CODE REVIEW

**Comment by churkin** [comment]
It is almost the same. These are a proxy (pipeline) errors and they are all associated with url.

**Comment by inikulin** [comment]
@churkin I mean we already have code which handles it, why not reuse it for consistency, why we need `switch/case` approach which is quite ugly and why we have methods in constant object `ERR`?

File: src/pipeline/errs.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        case originServerConnectionTerminated:
+            return `Failed to perform a request for the "${url}" resource because origin server connection was unexpectedly terminated.`;
+        case cantResolveUrl:
+            return `Failed to find DNS-record for the "${url}" resource.`;
+        case originServerRequestTimeout:
+            return `The server that hosts the tested page at ${url} did not respond to a connection request within ` +
+                   'the timeout period. The problem may be related to local machine\'s network or firewall settings, ' +
+                   'server outage, or network problems that make the server inaccessible.';
+        default:
+            throw new Error('Trying to receive the message for an unknown error.');
```
Comment (VasilyStrelyaev) [inline src/pipeline/errs.js]: Attempted to...

File: src/pipeline/errs.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export const originServerConnectionTerminated = 'originServerConnectionTerminated';
+export const cantResolveUrl                   = 'cantResolveUrl';
+export const originServerRequestTimeout       = 'originServerRequestTimeout';
+
+export function getMsg (err, url) {
+    switch (err) {
+        case originServerConnectionTerminated:
+            return `Failed to perform a request for the "${url}" resource because origin server connection was unexpectedly terminated.`;
+        case cantResolveUrl:
+            return `Failed to find DNS-record for the "${url}" resource.`;
```
Comment (VasilyStrelyaev) [inline src/pipeline/errs.js]: Failed to find a DNS-record for the resource at "${url}".

File: src/pipeline/errs.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export const originServerConnectionTerminated = 'originServerConnectionTerminated';
+export const cantResolveUrl                   = 'cantResolveUrl';
+export const originServerRequestTimeout       = 'originServerRequestTimeout';
+
+export function getMsg (err, url) {
+    switch (err) {
+        case originServerConnectionTerminated:
+            return `Failed to perform a request for the "${url}" resource because origin server connection was unexpectedly terminated.`;
```
Comment (VasilyStrelyaev) [inline src/pipeline/errs.js]: Failed to perform a request for the resource at "${url}" because connection to an origin server was unexpectedly terminated.

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by churkin** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by kirovboris** [comment]
lgtm

**Comment by LavrovArtem** [comment]
lgtm

**Comment by inikulin** [comment]
\r-

**Comment by inikulin** [comment]
lgtm

**Pull request merged**